src/confab/agent/runner.go

```diff
@@ -58,12 +61,19 @@ func (r *Runner) Run() error {
 		return err
 	}
 
-	go r.cmd.Wait() // reap child process if it dies
+	r.wg.Add(1)
+	go func() {
+		r.cmd.Wait()
+		atomic.StoreInt32(&r.exited, 1)
+		r.wg.Done()
+	}()
 
 	r.Logger.Info("agent-runner.run.success")
 	return nil
 }
 
+func (r *Runner) Exited() bool { return atomic.LoadInt32(&r.exited) == 1 }
```

Comment:
Instead of using an atomic value for keeping track of the process state, why don't we directly query it using these functions: 
https://github.com/cloudfoundry-incubator/consul-release/pull/24/files#diff-af3611f3fdf88edb580d3ab25bb2cb63R10
https://github.com/cloudfoundry-incubator/consul-release/pull/24/files#diff-ce4f157bcc7324fc16485f8822c9357fR7

Removing the `processIsRunning` helper function in favor of an atomic value, is a weaker assertion against the processes actual state. Especially since we have os dependent utils which will give us definitive answers as to the processes current state.